x-pack/legacy/plugins/ml/server/routes/anomaly_detectors.ts

```diff
@@ -0,0 +1,460 @@
+   *
+   * @apiSuccess {Number} count
+   * @apiSuccess {Object[]} jobs
+   */
+  router.get(
+    {
+      path: '/api/ml/anomaly_detectors/_stats',
+      validate: false,
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.jobStats');
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {get} /api/ml/anomaly_detectors/:jobId/_stats Get stats for requested anomaly detection job
+   * @apiName GetAnomalyDetectorsStatsById
+   * @apiDescription Returns anomaly detection job statistics.
+   *
+   * @apiParam {String} jobId Job ID.
+   */
+  router.get(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}/_stats',
+      validate: {
+        params: schema.object({
+          jobId: schema.string(),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { jobId } = request.params;
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.jobStats', { jobId });
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {put} /api/ml/anomaly_detectors/:jobId Instantiate an anomaly detection job
+   * @apiName CreateAnomalyDetectors
+   * @apiDescription This API creates an anomaly detection job that performs an analysis
+   *                 on the source index and stores the outcome in a destination index.
+   *
+   * @apiParam {String} jobId Job ID.
+   */
+  router.put(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}',
+      validate: {
+        params: schema.object({
+          jobId: schema.string(),
+        }),
+        body: schema.object({ ...anomalyDetectionJobSchema }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { jobId } = request.params;
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.addJob', {
+          jobId,
+          body: request.body,
+        });
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {put} /api/ml/anomaly_detectors/:jobId/_update Update an anomaly detection job
+   * @apiName UpdateAnomalyDetectors
+   * @apiDescription This API updates an anomaly detection job
+   *
+   * @apiParam {String} jobId Job ID.
+   */
+  router.post(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}/_update',
```

Comment:
If I edit a job (add a job description, or add to a group) I see the following error:

![image](https://user-images.githubusercontent.com/7405507/73764222-7b5cb480-476a-11ea-8a40-da8d3726f06c.png)

with the browser console showing a 404 error `{statusCode: 400, error: "Bad Request", message: "[request body.custom_settings.created_by]: definition for this key is missing"}`

Reply:
Good spot! 😄 Added `created_by` definition to `custom_settings` in the schema in b70059756cf4d0eb6a125b0ba7e891c69daa622d